packages/jest-snapshot/src/SnapshotFile.js

```diff
@@ -43,11 +45,22 @@ class SnapshotFile {
   }
 
   save() {
+    const operationResult = {};
+    this._uncheckedKeys.forEach(key => delete this._content[key]);
```

Comment:
I think this could benefit from a comment, like:

// Remove stale keys from the snapshot.